Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx

```diff
@@ -155,37 +177,114 @@ enum TrackPairs {
 };
 } // namespace correlationstask
 
-SystemType getSystemType()
+/// \brief System type according to configuration string
+/// \param sysstr The system configuration string
+/// \return The internal code for the passed system string
+SystemType getSystemType(std::string const& sysstr)
 {
   /* we have to figure out how extract the system type */
+  if (sysstr.empty() or (sysstr.compare("PbPb") == 0)) {
+    return kPbPb;
+  } else if (sysstr.compare("pp") == 0) {
+    return kpp;
+  } else if (sysstr.compare("pPb") == 0) {
+    return kpPb;
+  } else if (sysstr.compare("Pbp") == 0) {
+    return kPbp;
+  } else if (sysstr.compare("pPb") == 0) {
+    return kpPb;
+  } else if (sysstr.compare("XeXe") == 0) {
```

Comment:
Why not just `sysstr == "abcd"`?

Reply:
You are right!
I guess I repeated what was needed when it is embedded within a `Configurable` but the cast in the function signature transforms it in a conventional `std::string`.
Thanks for pointing that out!
I modified the code accordingly